server/sonar-server/src/main/java/org/sonar/server/component/NewComponent.java

```diff
@@ -19,14 +19,17 @@
  */
 package org.sonar.server.component;
 
-import com.google.common.base.Preconditions;
-import org.sonar.api.resources.Qualifiers;
-
 import javax.annotation.CheckForNull;
 import javax.annotation.Nullable;
+import org.sonar.api.resources.Qualifiers;
+
+import static com.google.common.base.Preconditions.checkArgument;
 
 public class NewComponent {
 
+  private static final int MAX_KEY_LENGHT = 400;
+  private static final int MAX_NAME_LENGTH = 2000;
```

Comment:
Creating a project with a name size more then 400 generates a 500 error (because in resource_index, size is limited to 400)